src/main/java/io/spokestack/spokestack/Spokestack.java

```diff
@@ -0,0 +1,659 @@
+         */
+        public Builder setConfig(SpeechConfig config) {
+            this.pipelineBuilder.setConfig(config);
+            this.nluBuilder.setConfig(config);
+            this.ttsBuilder.setConfig(config);
+            return this;
+        }
+
+        /**
+         * Sets a configuration value.
+         *
+         * @param key   Configuration property name
+         * @param value Property value
+         * @return the updated builder
+         */
+        public Builder setProperty(String key, Object value) {
+            this.speechConfig.put(key, value);
+            return this;
+        }
+
+        /**
+         * Sets a transcript editor used to alter ASR transcripts before they
+         * are classified by the NLU subsystem.
+         *
+         * <p>
+         * If a transcript editor is in use, registered listeners will receive
+         * {@code RECOGNIZE} events from the speech pipeline with the unedited
+         * transcripts, but the editor will automatically run on those
+         * transcripts before the NLU subsystem operates on them. Thus, the
+         * {@code utterance} inside the {@code NLUResult} returned by
+         * classification will reflect the edited version of the transcript.
+         * </p>
+         *
+         * <p>
+         * Transcript editors are <i>not</i> automatically run on inputs to the
+         * {@link #classify(String)} convenience method.
+         * </p>
+         *
+         * @param editor A transcript editor used to alter ASR results before
+         *               NLU classification.
+         * @return the updated builder
+         */
+        public Builder withTranscriptEditor(TranscriptEditor editor) {
+            this.transcriptEditor = editor;
+            return this;
+        }
+
+        /**
+         * Sets the Android Context for the pipeline. Should be an Application
+         * Context rather than an Activity Context.
+         *
+         * @param androidContext the Android Application Context.
+         * @return the updated builder
+         */
+        public Builder withAndroidContext(Context androidContext) {
+            this.appContext = androidContext;
+            this.pipelineBuilder.setAndroidContext(androidContext);
+            this.ttsBuilder.setAndroidContext(androidContext);
+            return this;
+        }
+
+        /**
+         * Sets the Android Lifecycle used for management of TTS playback.
+         *
+         * @param lifecycle the Android Lifecycle.
+         * @return the updated builder
+         */
+        public Builder withLifecycle(Lifecycle lifecycle) {
+            this.appLifecycle = lifecycle;
+            this.ttsBuilder.setLifecycle(lifecycle);
+            return this;
+        }
+
+        /**
+         * Signal that Spokestack's speech pipeline should not be used to
+         * recognize speech.
+         *
+         * @return the updated builder
+         */
+        public Builder disableAsr() {
```

Comment:
This should be renamed, as it does not just disable ASR. Maybe `disableSpeechPipeline`?

Reply:
ASR is the purpose of the speech pipeline. Wakeword is just a gatekeeper for ASR; it doesn't make sense to have one without the other.

Reply:
> Wakeword is just a gatekeeper for ASR; it doesn't make sense to have the former without the latter.
Disagree here, there's a use case for a wakeword activation that does not initiate asr. Remember "party time"?

Reply:
But more broadly, this disables the entire speech pipeline, not just asr, and should be named to reflect that.

Reply:
I remember no parties. But sure; the name reflects that assumption, which also fails to consider things like `KeywordRecognizer`, so point conceded.